enterprise/backup/src/test/java/org/neo4j/backup/OnlineBackupCommandIT.java

```diff
@@ -142,6 +158,56 @@ public void makeSureBackupCanBePerformedWithCustomPort() throws Exception
         assertEquals( getDbRepresentation(), getBackupDbRepresentation( "customport" ) );
     }
 
+    @Test
+    public void onlyTheLatestTransactionIsKeptAfterIncrementalBackup() throws Exception
+    {
+        // given database exists with data
+        int port =  4446;
+        startDb( "" + port );
+        createSomeData( db );
+
+        // and we have a full backup
+        String backupName = "backupName" + recordFormat;
+        File backupLocation = new File( backupDir, backupName );
+        String address = "localhost:" + port;
+        assertEquals(
+                0,
+                runBackupToolFromOtherJvmToGetExitCode(
+                        "--from", address,
+                        "--cc-report-dir=" + backupDir,
+                        "--backup-dir=" + backupDir,
+                        "--name=" + backupName ) );
+
+        // and the database contains a few more transactions
+        long aMegabyte = ByteUnit.mebiBytes( 1 );
+        long numberOfBytesInTx = 100;
+        LongStream.range( 0, 205 ).forEach( number -> createSomeData( db, aMegabyte / numberOfBytesInTx ) );
```

Comment:
This does take longer than expected, however there is no quick fix to make this more reasonable; In order to do so would mean modifying the configuration of the embedded database which isn't necessarily what we want to. Additionally rotation is tied to the tx streaming batch size, which is hard coded.

Reply:
Increasing tx size = out of memory errors and overall slower;
Reducing required tx for rotation = too difficult for something that should be so simple
Faking trafic/data to simply meet requirements = not possible